ktor-utils/jvm/src/io/ktor/util/cio/Semaphore.kt

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2014-2020 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.util.cio
+
+import kotlinx.coroutines.sync.Semaphore
+
+@Deprecated(
+    "Ktor Semaphore is deprecated and will be removed in ktor 2.0.0. Consider using kotlinx.coroutines Semaphore instead.",
+    level = DeprecationLevel.WARNING,
+    replaceWith = ReplaceWith("Semaphore", "kotlinx.coroutines.sync.Semaphore")
+)
+class Semaphore(val limit: Int) {
+    private val delegate = Semaphore(limit)
+
+    @Deprecated(
+        "Ktor Semaphore is deprecated and will be removed in ktor 2.0.0. Consider using kotlinx.coroutines Semaphore instead.",
+        level = DeprecationLevel.WARNING,
+        replaceWith = ReplaceWith("acquire()", "kotlinx.coroutines.sync.Semaphore")
```

Comment:
What it will produce on existing code? Doesn't IDEA generate mess?

Reply:
Just checked. Everything works fine. Could you record the screencast if it doesn't work for you?